crates/tags/src/storage/mod.rs

```diff
@@ -22,4 +22,27 @@ pub trait Storage: Send + Sync {
     async fn put(&self, tag: Tag, hash: Hash) -> Result<(), Self::Error>;
 }
 
-// TODO: Add tests
+#[cfg(test)]
+mod test {
+    use super::{Hash, Memory, Storage, Tag};
+
+    use std::ops::Deref;
+    use std::str::FromStr;
+
+    const HASH: &str = "sha256:LCa0a2j_xo_5m0U8HTBBNBNCLXBkg7-g-YpeiGJm564";
+
+    #[async_std::test]
+    async fn basic() {
+        let m = Memory::default();
+        assert_eq!(0, m.tags().await.unwrap().len());
+
+        let tag = Tag::from_str("1.2.3").unwrap();
+        let hash = Hash::from_str(HASH).unwrap();
+
+        m.put(tag.clone(), hash).await.unwrap();
+        let hash_retrieved = m.get(tag.clone()).await.unwrap();
+
+        assert_eq!(HASH, hash_retrieved.deref().to_string());
+        assert_eq!(1, m.tags().await.unwrap().len());
```

Comment:
```suggestion
        assert_eq!(vec![hash], m.tags().await.unwrap());
```

Don't just test for size. Test for contents too.